promises.ts

```diff
@@ -8,11 +8,42 @@
 
 export class UnexpectedResolveError extends Error {}
 
-export async function neverResolve<T>(promise: Promise<T>) {
-  await promise
-  throw new UnexpectedResolveError('promise resolved unexpectedly')
+export function sleep(millis: number, signal?: AbortSignal) {
+  return new Promise<void>((resolve) => {
+    function done() {
+      clearTimeout(timeoutHandle)
+      signal?.removeEventListener('abort', done)
+      resolve()
+    }
+    signal?.addEventListener('abort', done)
+    const timeoutHandle = setTimeout(done, millis)
+  })
 }
 
-export function sleep(millis: number) {
-  return new Promise((resolve) => setTimeout(resolve, millis))
+export function periodically(
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  func: () => Promise<any>,
```

Comment:
Would it be happy with this?

```suggestion
  func: () => Promise<unknown>,
```

or this?

```suggestion
  func: () => Promise<void>,
```

Reply:
Yea, going with the unknown option